tests/integration/meshexp/meshexp_test.go

```diff
@@ -0,0 +1,154 @@
+//  Copyright 2019 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+// How to run this test suite locally:
+// vmconfig="{
+// 	'vm_namespace': 'default',
+// 	'project_number': '895429144602',
+// 	'project_id': 'jianfeih-test',
+// 	'gcp_vm_zone': 'us-central1-a',
+// 	'gke_cluster_name': 'istio-dev'
+// }"
+// go test -v ./tests/integration/meshexp
+// -istio.test.env  kube -istio.test.hub "gcr.io/istio-release" \
+// -istio.test.tag "master-latest-daily" \
+// --log_output_level=tf:debug,CI:debug \
+// -istio.test.kube.meshexp.debianUrl=https://storage.googleapis.com/istio-release/releases/1.1.3/deb  \
+// -istio.test.kube.meshexp.vmconfig=$vmconfig
+// -istio.test.nocleanup
+//
+// Package meshexp contains test suite for mesh expansion.
+package meshexp
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	old_framework "istio.io/istio/tests/e2e/framework"
+
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/log"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/rawvm"
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+var (
+	vmInstance rawvm.Instance
+	// ports is the port the VM service exposes.
+	ports = model.PortList{
+		&model.Port{
+			Name:     "http",
+			Port:     8080,
+			Protocol: model.ProtocolHTTP,
+		},
+	}
+)
+
+const (
+	serviceName = ""
+)
+
+func TestMain(m *testing.M) {
```

Comment:
I don't think `integration` is the right place for this ... this is a heavy test that requires a lot of custom setup. I suspect it will have it's own target in CI.  Maybe a directory like `tests/vm` or `tests/gce` might be more appropriate? The latter is probably better in the sense that it quickly identifies the test as gce-specific.

@ozevren @louiscryan  @duderino  thoughts?

Reply:
it's a heavy test...because the setup process itself is heavy, not particular to gce.

Once we have other ci system (kind, eg) support vm-like thing, we can have vm test run there. that's why i make the vmconfig as opaque json string encoded.

the setup theoretically should also work for other environments, we need to define and test a few knobs to replace `gcloud compute ssh`.